src/guide-dist.md

```diff
@@ -10,47 +10,137 @@ pub trait Distribution<T> {
+
+-   [`Standard`] requires no parameters and samples values uniformly according
+    to the type. [`Rng::gen`] provides a short-cut to this distribution.
+-   [`Uniform`] is parametrised by `Uniform::new(low, high)` (including `low`,
+    excluding `high`) or `Uniform::new_inclusive(low, high)` (including both),
+    and samples values uniformly within this range.
+    [`Rng::gen_range`] is a convenience method defined over
+    [`Uniform::sample_single`], optimised for single-sample usage.
+-   [`Alphanumeric`] is uniform over the `char` values `0-9A-Za-z`.
+-   [`Open01`] and [`OpenClosed01`] are provide alternate sampling ranges for
+    floating-point types (see below).
+
+## Uniform sampling by type
+
+Lets go over the distributions by type:
+
+-   For `bool`, [`Standard`] samples each value with probability 50%.
+-   For `Option<T>`, the [`Standard`] distribution samples `None` with
+    probability 50%, otherwise `Some(value)` is sampled, according to its type.
+-   For integers (`u8` through to `u128`, `usize`, and `i*` variants),
+    [`Standard`] samples from all possible values while
+    [`Uniform`] samples from the parameterised range.
+-   For `NonZeroU8` and other "non-zero" types, [`Standard`] samples uniformly
+    from all non-zero values (rejection method).
+-   `Wrapping<T>` integer types are sampled as for the corresponding integer
+    type by the [`Standard`] distribution.
+-   For floats (`f32`, `f64`),
+
+    -   [`Standard`] samples from the half-open range `[0, 1)` with 24 or 53
+        bits of precision (for `f32` and `f64` respectively)
+    -   [`OpenClosed01`] samples from the half-open range `(0, 1]` with 24 or
+        53 bits of precision
+    -   [`Open01`] samples from the open range `(0, 1)` with 23 or 52 bits of
+        precision
+    -   [`Uniform`] samples from a given range with 23 or 52 bits of precision
+-   For the `char` type, the [`Standard`] distribution samples from all
+    available Unicode code points, uniformly; many of these values may not be
+    printable (depending on font support). The [`Alphanumeric`] samples from
+    only a-z, A-Z and 0-9 uniformly.
+-   For tuples and arrays, each element is sampled as above, where supported.
+    The [`Standard`] and [`Uniform`] distributions each support a selection of
+    these types (up to 12-tuples and 32-element arrays).
+    This includes the empty tuple `()` and array.
+-   For SIMD types, each element is sampled as above, for [`Standard`] and
+    [`Uniform`] (for the latter, `low` and `high` parameters are *also* SIMD
+    types, effectively sampling from multiple ranges simultaneously). SIMD
+    support is gated behind a [feature flag](../features.html#simd-support).
+
+# Non-uniform distributions
+
+Non-uniform distributions can be divided into two categories, as follows.
+Some of these discrete and all of the continuous distributions have been moved
+from the main [`rand`] crate to a dedicated [`rand_distr`] crate.
```

Comment:
I'm not sure how relevant it is to mention the past, maybe it's better to just say where they are?

Reply:
Currently these still appear as deprecated in the main crate.

Reply:
Fair enough. I guess we can rewrite that section once they are removed for good.